api/addon/v1alpha1/addon_types.go

```diff
@@ -409,13 +430,11 @@ func (a *Addon) SetPrereqStatus(phase ApplicationAssemblyPhase, reasons ...strin
 	switch phase {
 	case Pending:
 		a.Status.Lifecycle.Prereqs = Pending
-		a.SetInstallStatus(Pending)
 	case Failed:
 		a.Status.Lifecycle.Prereqs = Failed
-		a.SetInstallStatus(Failed, reasons...)
+		a.Status.Reason = strings.Join(reasons, ", ")
```

Comment:
Just setting reason here will conflict w/ install status, we also do not currently ever care about prereqs status, it's merely a status set but all state of the Addon is only reflected by the singular `install` status, so setting `prereqs` must also set the `install` status.

I think we shouldn't be setting reason directly outside of the helper addon methods.